idearium-lib/lib/query.js

```diff
@@ -0,0 +1,61 @@
+'use strict';
+
+/**
+ * Assign some defaults to the passed Mongoose query options.
+ * @param {Object} options Mongoose query options.
+ * @return {Object} Returns the options object for Mongoose queries.
+ */
+const getOptions = (options) => {
+
+    const defaults = {
+        filter: {},
+        lean: true,
+        limit: 10,
+        projection: '',
+    };
+
+    Object.assign(defaults, options);
+
+    // We want to return all fields.
+    if (options.projection === false) {
+        delete defaults.projection;
+    }
+
+    return defaults;
+
+};
+
+/**
+ * Find documents from a model.
+ * @param {Object} model Mongoose model.
+ * @param {Object} options Mongoose query options.
+ * @return {Promise} Returns a Mongoose query.
+ */
+const find = (model, options) => {
+
+    const { filter, lean, limit, projection } = getOptions(options);
+
+    return model.find(filter, projection)
+        .lean(lean)
+        .limit(limit)
+        .exec();
```

Comment:
I think executing should be beyond the scope of the function. This will also allow us to more easily run tests. This should be a pure function and do one thing, given options, generate a Mongoose query object. That's it. Not run it to.